.github/workflows/scripts/validate-form.py

```diff
@@ -155,7 +165,132 @@ def validate_form(issue_content_file: str) -> None:
             )
         )
 
-    # TODO: legistar
+    # Test Legistar
+    legistar_response = None
+    if (
+        form_values[LEGISTAR_CLIENT_ID] is not None
+        and form_values[LEGISTAR_CLIENT_TIMEZONE] is not None
+    ):
+        log.info("Attempting Legistar data retrieval")
+
+        # Create Scraper and fill with get time zone function impl
+        class CustomScraper(LegistarScraper):
+            def get_time_zone(self) -> str:
+                return form_values[LEGISTAR_CLIENT_TIMEZONE]
+
+        # Init temp scraper and run
+        scraper = CustomScraper(client=form_values[LEGISTAR_CLIENT_ID])
+        try:
+            # Check that the provided client information is even a Legistar municipality
+            if not scraper.is_legistar_compatible:
+                legistar_response = (
+                    f":x: No public Legistar instance found for "
+                    f"the provided client ({form_values[LEGISTAR_CLIENT_ID]}). "
+                    f"If your municipality uses Legistar but you received this error, "
+                    f"we recommended contacting your municipality clerk and asking "
+                    f"about public Legistar API access, they may direct you to the IT "
+                    f"department as well."
+                )
+            log.info("Legistar client available")
+
+            # If everything runs correctly, log success and show event
+            # model in comment
+            for days_prior in [7, 14, 28]:
+                log.info(f"Attempting minimum CDP data for {days_prior} days")
+                if scraper.check_for_cdp_min_ingestion(check_days=days_prior):
+                    log.info("Legistar client has minimum data")
+                    events = scraper.get_events(
+                        begin=datetime.utcnow() - timedelta(days=days_prior),
+                    )
+                    single_event = events[0].to_dict()
+                    event_as_json_str = json.dumps(single_event, indent=4)
+
+                    legistar_response = (
+                        f":heavy_check_mark: The municipality's Legistar instance "
+                        f"contains the minimum required CDP event ingestion data.\n"
+                        f"<summary>Retrieved Data</summary>\n"
+                        f"<details>\n\n"  # Extra new line for proper rendering
+                        f"```json\n"
+                        f"{event_as_json_str}\n"
+                        f"```"
+                        f"</details>"
+                    )
+                    break
+
+            if legistar_response is None:
+                log.info(
+                    "Legistar client missing minimum data, "
+                    "attempting to simply pull data for logging."
+                )
+                # Check if _any_ data was returned
+                for days_prior in [7, 14, 28]:
+                    log.info(f"Attempting to pull Legistar data for {days_prior} days")
```

Comment:
Might be nice to include "previous" in this sentence just to be abundantly clear.

"Attempting to pull Legistar data for the previous {days_prior} days."